_posts/en/newsletters/2025-12-19-newsletter.md

```diff
@@ -0,0 +1,1246 @@
+  tapscript could be combined with [BIP360][] to fully quantum harden Bitcoin
+  outputs.
+
+  Tadge Dryja [proposed][news qr agg] one way in which bitcoin could implement
+  general cross-input signature aggregation which would partially mitigate the
+  large size of post-quantum signatures.
+
+</div>
+
+## May
+
+{:#clustermempool}
+- **Cluster mempool:**
+  In January, Stefan Richter prompted excitement by [discovering][news340 richter ggt]
+  that an efficient algorithm for the _maximum-ratio closure problem_ from a
+  1989 research paper could be applied to cluster linearization. Pieter Wuille
+  extended his research to incorporate this minimal-cut-based third approach.
+  In February, Pieter Wuille [walked][news341 pr-review-club txgraph] the Bitcoin Core
+  PR Review Club through the newly introduced `TxGraph` class which distills
+  transactions to only weight, fees, and relationships for efficient
+  interaction with the mempool graph.
+  In May, Wuille described the [tradeoffs][news352 wuille linearization techniques] of
+  the three cluster linearization approaches and provided benchmarks, finding
+  that both advanced approaches were far more efficient than the simple
+  candidate-set search, but the linear programming-based spanning-forest
+  linearization algorithm would be more practical than the min-cut-based
+  algorithm.
+  In October, Abubakar Sadiq Ismail [described][news377 ismail
+  template improvement] how the cluster mempool could be leveraged to track
+  when the mempool content had significantly improved upon a prior block
+  template.
+  In November, the cluster mempool implementation was [completed][news382 cluster
+  mempool completed], staging it to be released per Bitcoin
+  Core 31.0. Work to replace the initial candidate-set search linearization
+  algorithm with the spanning-forest linearization algorithm is on-going.
+
+{:#opreturn}
+- **Increasing or removing Bitcoin Core’s OP_RETURN size limit:** In April,
+  protocol developers discovered that the OP_RETURN output limits caused a
+  malincentive to embed data in payment outputs under some circumstances. In
+  addition to the observation that the policy’s intended benefits had been eroded
+  by the network’s growth, this prompted a proposal to drop the OP_RETURN mempool
+  policy limits. This proposition kicked off a heated debate about the efficacy
+  of mempool policy, the purpose of Bitcoin, and the responsibility of Bitcoin
+  developers to regulate or refrain from regulating usage of Bitcoin. Bitcoin
+  Core contributors argued that economic incentives made it unlikely that
+  OP_RETURN outputs would see drastically more use and considered the change to be
+  fixing the incentive bug. Critics interpreted the removal of the limits as an
+  endorsement of data embedding, but also agreed that it is economically
+  unattractive to be used that way. Eventually, the Bitcoin Core 30.0 release
```

Comment:
Could, optionally, link to https://bitcoinops.org/en/newsletters/2025/06/13/#bitcoin-core-32406 here

Reply:
Oh oops, I overlooked this comment earlier. Will address it now.